server/src/main/java/io/crate/types/ResultSetParser.java

```diff
@@ -80,11 +80,16 @@ public static Object getObject(ResultSet resultSet, int i, String typeName) thro
                 return PgOidVectorType.listFromOidVectorString(textval);
             }
             case "char":
-                String strValue = resultSet.getString(columnIndex);
-                if (strValue == null) {
+                // It's NOT char(1) which is actually "character".
+                // It's char with OID 18, ie single byte "char" which can store single ASCII character.
+                // See also https://www.postgresql.org/docs/15/datatype-character.html#DATATYPE-CHARACTER-SPECIAL-TABLE
+                byte[] charBytes = resultSet.getBytes(columnIndex);
+                if (charBytes == null) {
                     return null;
                 }
-                return Byte.valueOf(strValue);
+                String str = new String(charBytes, StandardCharsets.US_ASCII);
+                assert str.length() == 1 : "Only single ASCII character is allowed";
+                return str;
```

Comment:
Can we just return `resultSet.getString(columnIndex)` here?
I don't think we need to do additional validation, that's up to the foreign server to ensure that a char type has a value matching its char type semantics

Reply:
updated